packages/falso/src/lib/core/core.ts

```diff
@@ -1,40 +1,116 @@
 import { random } from '../random';
+import { primitiveValueIsUnique } from './unique-validators';
 
 export interface FakeOptions {
   length?: number;
+  priority?: 'length' | 'unique';
   locale?: any | string[];
 }
 
 export type markRequired<Type, Key extends keyof Type> = Type & {
   [Property in Key]-?: Type[Property];
 };
 
+export interface FakeConfig<T> {
+  uniqueComparer: (
+    item: T,
+    items: T[],
+    comparisonKeys?: (keyof T)[]
+  ) => boolean | ((item: T, items: T[]) => boolean);
+  comparisonKeys?: (keyof T)[];
+}
+
 export type Return<T, O extends FakeOptions> = [O] extends [never]
   ? T
   : O['length'] extends number
   ? T[]
   : T;
 
-type FactoryFunction<T> = (i: number) => T;
+export type FactoryFunction<T> = (i: number) => T;
 
 export function fake<T, Options extends FakeOptions>(
   data: Readonly<T[]> | FactoryFunction<T>,
-  options?: Options
+  options?: Options,
+  config: FakeConfig<T> = {
+    uniqueComparer: primitiveValueIsUnique,
+  }
 ): Return<T, Options> {
-  const dataSource = Array.isArray(data)
-    ? () => randElement(data)
-    : (data as FactoryFunction<T>);
+  if (options?.length === 0) {
+    return [] as any;
+  }
+
+  if (Array.isArray(data)) {
+    return fakeFromArray(data, options) as any;
+  }
+
+  return fakeFromFunction(data as FactoryFunction<T>, config, options) as any;
+}
+
+export function fakeFromFunction<T, Options extends FakeOptions>(
+  data: FactoryFunction<T>,
+  config: FakeConfig<T>,
+  options?: Options
+): T | T[] {
+  if (!options?.length) {
```

Comment:
You are handling the length options both here and in the `fake` function, please choose one place